### PR TITLE
Solve nested unwind bug in compiled runtime and add some feature tests.

### PR DESCRIPTION
**Pull request by Lojjs**
To be forward merged to 3.3 and 3.4

### CODE REVIEW

File: community/cypher/cypher/src/main/java/org/neo4j/cypher/internal/codegen/CompiledConversionUtils.java [hunk 1]
```diff
@@ -416,30 +428,15 @@ else if ( iterable == null )
         {
             return Collections.emptyIterator();
         }
+        else if ( iterable instanceof String )
+        {
+            String[] singleStringArray = {(String) iterable};
```
Comment (sherfert) [inline community/cypher/cypher/src/main/java/org/neo4j/cypher/internal/codegen/CompiledConversionUtils.java]: You can use Stream.of( iterable ).iterator()

File: enterprise/cypher/acceptance-spec-suite/src/test/resources/cypher/features/UnwindAcceptance.feature [hunk 2]
```diff
@@ -228,3 +228,57 @@ Feature: UnwindAcceptance
...
+      | 3.5 |
+      | 4.5 |
+      | 5.5 |
+    And no side effects
+
+  Scenario: Nested unwind with strings
+    Given an empty graph
+    When executing query:
+      """
+      WITH [['a', 'b'], ['c', 'd'], 'e'] AS nested
```
Comment (craigtaverner) [inline enterprise/cypher/acceptance-spec-suite/src/test/resources/cypher/features/UnwindAcceptance.feature:271]: Should also work with mixed types

File: community/cypher/cypher/src/main/java/org/neo4j/cypher/internal/codegen/CompiledConversionUtils.java [hunk 3]
```diff
@@ -400,18 +401,35 @@ else if ( iterable instanceof PrimitiveEntityStream )
...
         }
+        else if ( iterable instanceof Integer)
+        {
+            return (IntStream.of((Integer) iterable)).iterator();
+        }
         else if ( iterable instanceof IntStream )
         {
             return ((IntStream) iterable).iterator();
         }
+        else if ( iterable instanceof String )
```
Comment (craigtaverner) [inline community/cypher/cypher/src/main/java/org/neo4j/cypher/internal/codegen/CompiledConversionUtils.java]: Is it necessary to have type specific checks here? Perhaps just `Stream.of(iterable)` in the `else` clause would be simpler? What does the interpreted runtime do in this case?

**Review by craigtaverner — CHANGES_REQUESTED** [review]

File: community/cypher/cypher/src/main/java/org/neo4j/cypher/internal/codegen/CompiledConversionUtils.java (hunk 3 above)
Comment (sherfert) [reply community/cypher/cypher/src/main/java/org/neo4j/cypher/internal/codegen/CompiledConversionUtils.java]: Usually the primitive streams are more performant. But then again, a stream with just one element won't be a bottleneck...

File: enterprise/cypher/acceptance-spec-suite/src/test/resources/cypher/features/UnwindAcceptance.feature (hunk 2 above)
Comment (Lojjs) [reply enterprise/cypher/acceptance-spec-suite/src/test/resources/cypher/features/UnwindAcceptance.feature:271]: Good point, I have added a new test for that.

**Review by craigtaverner — APPROVED** [review]

**Pull request merged**